### PR TITLE
Unify soft_pick temperature cutoff code

### PR DESCRIPTION
**Pull request by sethtroisi**
both codebases now set temp_threshold to -1 if not using softpick.

both now do 0.98 squash (previously only c++ code did)

### CODE REVIEW

File: cc/mcts_player.cc [hunk 1]
```diff
@@ -164,7 +165,7 @@ Coord MctsPlayer::SuggestMove() {
 }
 
 Coord MctsPlayer::PickMove() {
-  if (!options_.soft_pick || root_->position.n() >= temperature_cutoff_) {
```
Comment (amj) [inline cc/mcts_player.cc]: with >=, doesn't that mean 30 is the first argmax?   i.e., black has one extra softpick move?  or is n indexed from 0?

Comment (sethtroisi) [reply cc/mcts_player.cc]: I 99% know it's indexed from 0 in Python. I assume it is in C++ too.

Comment (tommadams) [reply cc/mcts_player.cc]: >= is correct, it was originally incorrectly >

See PR #213

**Comment by amj** [comment]
LGTM ok2merge.

**Pull request merged**